### PR TITLE
Fixing the logic to get unprocessed messages from the azure event hub partition

### PR DESCRIPTION
**Pull request by SatishRanjan**
Fixes the issue described in #830
https://github.com/kedacore/keda/issues/830

### CODE REVIEW

**Comment by ahmelsayed** [comment]
Looks like there is a conflict with #830

**Comment by SatishRanjan** [comment]
> Looks like there is a conflict with #830

@ahmelsayed conflict with #830 is resolved now. Please take a look.

**Comment by christle** [comment]
Does it make sense to do the Check with the LastEnqueuedOffset before the Access to the storage Account?

**Comment by SatishRanjan** [comment]
> Does it make sense to do the Check with the LastEnqueuedOffset before the Access to the storage Account?

Yes, that optimization make sense. I'll re-order that line and return an empty checkpoint in that case.

File: pkg/scalers/azure_eventhub_scaler.go [hunk 1]
```diff
@@ -127,15 +127,31 @@ func (scaler *AzureEventHubScaler) GetUnprocessedEventCountInPartition(ctx conte
...
-		unprocessedEventCountInPartition = (math.MaxInt64 - partitionInfo.LastSequenceNumber) + checkpoint.SequenceNumber
+	//If checkpoint.Offset is empty that means no messages has been processed from an event hub partition
+	// And since partitionInfo.LastSequenceNumber = 0 for the very first message hence
+	// total unprocessed message will be partitionInfo.LastSequenceNumber + 1
+	if checkpoint.Offset == "" {
+		unprocessedEventCountInPartition = partitionInfo.LastSequenceNumber + 1
+		return unprocessedEventCountInPartition, checkpoint, nil
 	}
+
+	if partitionInfo.LastSequenceNumber == checkpoint.SequenceNumber || partitionInfo.LastSequenceNumber > checkpoint.SequenceNumber {
```
Comment (ahmelsayed) [inline pkg/scalers/azure_eventhub_scaler.go:143]: can be simplified to `>=`

Comment (SatishRanjan) [reply pkg/scalers/azure_eventhub_scaler.go:143]: yes :)

**Comment by SatishRanjan** [comment]
@christle @ahmelsayed  - both of the comments are resolved now. Please take a look

**Comment by ahmelsayed** [comment]
@SatishRanjan I'm not seeing a DCO in the commits, though the bot doesn't seem to mind for some reason. Can you make sure to add that to the commits so it's there when I merge them?

**Pull request merged**